### ISSUE TITLE
[Accessibility]: Screen reader do not announce when user is in the "Search workspace" dialog.

### ISSUE THREAD

**Issue by DBowen33** [microsoft/vscode#277170]
PROBLEM:
Screen readers are not giving any indication that the "Search workspace" dialog has been opened.

OBSERVED RESULTS:
When a user triggers the "Search workspace" dialog, there is no indication from screen readers that a dialog has been opened.

EXPECTED RESULTS:
Screen Reader should announce that the user is in a dialog as well as the accessible name of the dialog.

USER IMPACT:
Keyboard users will not be able to understand whether they are in a dialog for "Search workspace" or part of the entire web page.

STEPS TO REPRODUCE:
1. Open VSCode
2. Turn screen reader on
3. With the keyboard, press ENTER or SPACE while focused on "Search workspace" field at top of UI
4. Notice how there is no indication that the user is in a dialog

GIF of Issue: 

![Image](https://github.com/user-attachments/assets/d94a9078-2620-411a-81fa-95d52eeab71a)

ENVIRONMENT DETAILS:
Machine: Chromebook
Browser: Version 136.0.7103.150 (Official Build) (64-bit)
Accessibility Tool: ChromeVox.
Bug Filling Time: 20 mins.

Machine: MacBook Pro
Browser: Version 137.0.7151.104 (Official Build) (arm64)
Accessibility Tool: VoiceOver.
Bug Filling Time: 20 mins."

**Comment by meganrogge** [comment]
Can you pls attach a gif showing the issue?
